daemon/image_delete.go

```diff
@@ -137,6 +137,9 @@ func (daemon *Daemon) imgDeleteHelper(name string, list *[]types.ImageDelete, fi
 }
 
 func (daemon *Daemon) canDeleteImage(imgID string, force bool) error {
+	if daemon.Graph().IsHeld(imgID) {
```

Comment:
If the image gets held after this point, can it still be deleted? The issue with this test is that after it returns false, the condition may be true immediately after. You may need to hold a lock during this method to ensure a pull doesn't proceed until method has done its damage.